piccolo_admin/endpoints.py

```diff
@@ -680,6 +687,21 @@ def __init__(
                             ),
                         )
 
+        #######################################################################
+        # MFA
+
+        if mfa_provider:
+            private_app.mount(
+                path="/mfa-setup/",
+                app=RateLimitingMiddleware(
+                    app=mfa_setup(
+                        provider=mfa_provider,
+                        auth_table=self.auth_table,
+                    ),
+                    provider=rate_limit_provider,
+                ),
+            )
```

Comment:
Looks like the ratelimit provider is set to a more strict version if it's not provided in a few lines time. Can we also implement that here but with an even stricter ratelimit possible?

Some links from a reputable company adjacent to mine that popped up on my feed today as a throwback:
- Brute Forcing TOTP Multi-Factor Authentication is Surprisingly Realistic - [Link](https://pulsesecurity.co.nz/articles/totp-bruting)
- Supposing I want to brute force a numeric code? - [Link](https://totp.sova.st/)
```suggestion
        if mfa_provider:
               if not rate_limit_provider:
                   rate_limit_provider = InMemoryLimitProvider(
                       limit=5, timespan=60, block_duration=300
                   )

            private_app.mount(
                path="/mfa-setup/",
                app=RateLimitingMiddleware(
                    app=mfa_setup(
                        provider=mfa_provider,
                        auth_table=self.auth_table,
                    ),
                    provider=rate_limit_provider,
                ),
            )
```

Reply:
Great catch!

Thanks for the link - it was an interesting read.

I've set the rate limiting to 5 requests in 5 minutes. But then I was thinking about it some more, and the `mfa-setup` endpoint is just for setting up MFA, and not for logging in. The `session_login` endpoint handles that, and uses the main `rate_limit_provider` - so might have to make the defaults on that stricter, or do some kind of account lock out.

Reply:
Yea, something to consider anyway.

We should also likely change the session login so it's not a permanent rate limit by default